src/libsyntax/parse/parser.rs

```diff
@@ -6630,6 +6630,22 @@ impl<'a> Parser<'a> {
             self.expect(&token::OpenDelim(token::Brace))?;
             let mut trait_items = vec![];
             while !self.eat(&token::CloseDelim(token::Brace)) {
+                if let token::DocComment(_) = self.token {
+                    if self.look_ahead(1,
+                    |tok| tok == &token::Token::CloseDelim(token::Brace)) {
+                        let mut err = self.diagnostic().struct_span_err_with_code(
+                            self.span,
+                            "found a documentation comment that doesn't document anything",
+                            DiagnosticId::Error("E0584".into()),
+                        );
+                        err.help("doc comments must come before what they document, maybe a \
+                            comment was intended with `//`?",
+                        );
+                        err.emit();
```

Comment:
For everything between 6636 and 6644 you can replace it with `self.span_fatal_err(self.prev_span, Error::UselessDocComment).emit();`, like in https://github.com/rust-lang/rust/blob/fcfc7963891db5c91035ea527ead2ef998d6781b/src/libsyntax/parse/parser.rs#L5290

Reply:
https://github.com/rust-lang/rust/pull/57333#discussion_r249254186

This is the reason I used `struct_span_err_with_code`. Do we not need that now?

Reply:
(indeed, this does not seem like it needs to be a fatal error; I concur with @saleemjaffer's choice to use a non-fatal error here.)